interview-questions.md

```diff
@@ -9,59 +9,61 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 1. What is `this` in JavaScript?
 
-  Your answer:
-
-  Researched answer:
+  Your answer: Is a keyword use inside a class
 
+  Researched answer: The JavaScript this keyword refers to the object it belongs to. It has different values depending on where it is used: In a method, this refers to the owner object. Alone, this refers to the global object. In a function, this refers to the global object.
+  `this` is a JavaScript  keyword that is uses to access information within a class.`this` refers to the object it belongs to.`this` can be use in a function to
 
 
 2. What is React? Why would you use it?
 
-  Your answer:
+  Your answer: React is a lot of different components that come al together and create an application.
 
-  Researched answer:
+  Researched answer: React is an open source JavaScript library that is used for building user interfaces specifically for single page applications. React allows developers to create large web applications that can change data, without reloading the page. The main purpose of React is to be fast, scalable, and simple. React is a component based where it takes in many small reusable components that are joined together to make an application. The main purpose of React is to be fast, scalable, and simple.
 
 
 
 3. Which lifecycle method is required in a React class component?
 
-  Your answer:
+  Your answer: The only thing I can think of is return but I don't think is a method.
 
-  Researched answer:
+  Researched answer: The render() method, render() is a React lifecycle method and is the only  required method within a class component in React.
 
 
 
 4. What is JSX? What is one notable difference between HTML and JSX?
 
-  Your answer:
+  Your answer: I think that JSX is an extension of JavaScript
 
-  Researched answer:
+  Researched answer: JSX is a syntax extension of JavaScript that is used within React to help describe what the user interface would look like. It is use as a visual aid when writing code for the user interface.
+  The different between JSX and HTML is that. HTML use the syntax class,
+  and JSX uses the syntax className.
 
 
 
 5. What is yarn? What file(s) are modified in a React application when you run yarn?
 
-  Your answer:
+  Your answer: Yarn is a package in JavaScript that manage node. Yarn add a node module file.
 
-  Researched answer:
+  Researched answer: Yarn is a JavaScript package manager that aims to be speedy, deterministic, and secure ,meaning it manages node module. Running yarn in our terminal will create a node module and a SRC folder into the computer. Running yarn installs some package of code from a global registry onto de developers local computer.
```

Comment:
Nice! Yes, yarn will connect dependencies. The yarn.lock, package.json, and node_nodules will be updated.